bin/echo.c

```diff
@@ -165,10 +214,12 @@ int echo(struct s2n_connection *conn, int sockfd)
     errno = 0;
 
     /* Act as a simple proxy between stdin and the SSL connection */
-    int p;
+    int p = 0;
     s2n_blocked_status blocked;
     do {
-        while ((p = poll(readers, 2, -1)) > 0) {
+        /* echo will send and receive Application Data back and forth between
+         * client and server until stop_echo is true. */
+        while (!(*stop_echo) && (p = poll(readers, 2, -1)) > 0) {
```

Comment:
This looping condition is getting kind of complex. Maybe just do a if(*stop_echo) { break; } in the loop?